arch/armv7a/memcpy.S

```diff
@@ -0,0 +1,361 @@
+#define DST r0
+#define SRC r1
+#define LEN r8
+#define DST_RET r12
+
+.arm
+.syntax unified
+
+.text
+.align 4
+
+/* void *memcpy(void *dst, const void *src, size_t l) */
+.globl memcpy
+.type memcpy, %function
+memcpy:
+	str r8, [sp, #-8]!
+	mov r8, r2
+	cmp LEN, #64
+	mov DST_RET, DST /* preserve return value */
```

Comment:
Perhaps it might be better to use r2 for `len` (no mov needed), preserve dst on stack and use r12 for temp register?

Reply:
I need even and then odd (in this order) registers for double-word copy, so if we want this to work also on armv7/8-m, we have to preserve `r9`, that's why I can't use r8 in d-word load/store.

Reply:
Oh, ok, makes sense